SpatialGDK/Source/SpatialGDK/Private/Tests/SpatialView/EntityComponentRecordTest.cpp

```diff
@@ -0,0 +1,187 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#include "Tests/TestDefinitions.h"
+
+#include "SpatialView/EntityComponentRecord.h"
+
+#include "EntityComponentTestUtils.h"
+
+#define ENTITYCOMPONENTRECORD_TEST(TestName) \
+	GDK_TEST(Core, EntityComponentRecord, TestName)
+
+namespace SpatialGDK
+{
+
+namespace
+{
+	const Worker_EntityId TEST_ENTITY_ID = 1337;
+	const Worker_ComponentId TEST_COMPONENT_ID = 1338;
+	const double TEST_VALUE = 7331;
+	const double TEST_UPDATE_VALUE = 7332;
+}  // anonymous namespace
+
+ENTITYCOMPONENTRECORD_TEST(GIVEN_empty_component_record_WHEN_component_added_THEN_has_component_data)
+{
+	// GIVEN
+	auto TestData = CreateTestComponentData(TEST_COMPONENT_ID, TEST_VALUE);
```

Comment:
I would check over for autos. This one can be replaced.

Reply:
Removed auto